File: tests/eval_c/451.c
#include <assert.h>
#include <string.h>

void f451(const char *s) {
    int i = 0;
    while (i < strlen(s)) {
        char a = s[i];
        char x = s[i + 1];
        char b = s[i + 2];
        
        assert(a == 'a');
        assert(b == 'b');
        assert(x != 'a' && x != 'b');
        
        i += 1;
    }
}

File: benchmarks/oopsla25/subjects/451.py
import sys

def f451(s: str):
  i = 0
  while i < len(s):
    a,x,b = s[i:i+3]
    assert a == "a"
    assert b == "b"
    assert x != "a" and x != "b"
    i += 3

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f451(my_string)
